common/web/types/src/kmx/kmx-file-reader.ts

```diff
@@ -0,0 +1,44 @@
+import { KMXFile, BUILDER_COMP_KEYBOARD, KEYBOARD, STORE } from "./kmx.js";
+import * as r from 'restructure';
+
+export class KmxFileReader {
+  private readonly rString = new r.String(null, 'utf16le');
+
+  private readString(source: Uint8Array, offset: number): string {
+    if(offset == 0) {
+      // the file has a null string here, which is not the same as an empty
+      // string ('')
+      return null;
```

Comment:
Is this ever an error to be flagged/logged?

Reply:
I guess it's up to the caller to handle?

Reply:
> I guess it's up to the caller to handle?

Yes, up to the caller. It's not an error in the file format.